Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -1634,6 +1637,8 @@ void WebPageProxy::loadDataWithNavigationShared(Ref<WebProcessProxy>&& process,
     process->assumeReadAccessToBaseURL(*this, baseURL);
     process->send(Messages::WebPage::LoadData(loadParameters), webPageID);
     process->startResponsivenessTimer();
+
+    updateNetworkIssueReporterIfNeeded();
```

Comment:
At what point do we split some of these things out into a shared function? Seems hairy to expect everyone to hit all the loading paths every time.

Reply:
Also is "update" really "createNew"?

Reply:
> At what point do we split some of these things out into a shared function? Seems hairy to expect everyone to hit all the loading paths every time.

Good question! Maybe we should have a new helper method like `prepareToLoadWebPage(LoadParameters&)`; we could then move this new logic into that method, along with a call to `addPlatformLoadParameters()`, which we currently call everywhere that we also `createNewNetworkIssueReporterIfNeeded`.

> Also is "update" really "createNew"?

I like that better too — renamed to `createNewNetworkIssueReporterIfNeeded()`.